test/cql-pytest/util.py

```diff
@@ -139,6 +143,50 @@ def new_secondary_index(cql, table, column, name='', extra=''):
     finally:
         cql.execute(f"DROP INDEX {keyspace}.{name}")
 
+# Helper function for establishing a connection with given username and password
+def cql_session(host, port, ssl, username, password):
+    profile = ExecutionProfile(
+        load_balancing_policy=RoundRobinPolicy(),
+        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
+        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
+        # The default timeout (in seconds) for execute() commands is 10, which
+        # should have been more than enough, but in some extreme cases with a
+        # very slow debug build running on a very busy machine and a very slow
+        # request (e.g., a DROP KEYSPACE needing to drop multiple tables)
+        # 10 seconds may not be enough, so let's increase it. See issue #7838.
+        request_timeout = 120)
+    if ssl:
+        # Scylla does not support any earlier TLS protocol. If you try,
+        # you will get mysterious EOF errors (see issue #6971) :-(
+        ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+    else:
+        ssl_context = None
+    cluster = Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
+        contact_points=[host],
+        port=int(port),
+        # TODO: make the protocol version an option, to allow testing with
+        # different versions. If we drop this setting completely, it will
+        # mean pick the latest version supported by the client and the server.
+        protocol_version=4,
+        auth_provider=PlainTextAuthProvider(username=username, password=password),
+        ssl_context=ssl_context,
+    )
+    return cluster.connect()
+
+@contextmanager
+def new_user(cql, username=''):
+    if not username:
+        username = unique_name()
+    cql.execute(f"CREATE ROLE {username} WITH PASSWORD = '{username}' AND LOGIN = true")
+    try:
+        yield username
+    finally:
+        cql.execute(f"DROP ROLE {username}")
+
+def new_session(cql, username):
+    endpoint = cql.hosts[0].endpoint
+    return cql_session(host=endpoint.address, port=endpoint.port, ssl=False, username=username, password=username)
```

Comment:
Please make this or cql_session a contextmanager as well, and in the finally, do cluster.shutdown() when the session ends. Otherwise it leaves around threads and who knows what.
In the cql fixture I forgot to do it because there's just one of them, but in the check_cql() function in test/cql-pytest/run.py, I didn't forget. 
We should also fix the cql fixture to yield, not return, and then call cluster.shutdown().